### PR TITLE
Espresso: important clean-up

### PR DESCRIPTION
**Pull request by tomdemeyere**
### Checklist

- [x] I have read the [Contributing Guide](https://quantum-accelerators.github.io/quacc/dev/contributing.html). Don't lie! 😉
- [x] My PR is on a custom branch and is _not_ named `main`.
- [x] I have used `black`, `isort`, and `ruff` as described in the [style guide](https://quantum-accelerators.github.io/quacc/dev/contributing.html#style).
- [ ] I have added relevant, comprehensive [unit tests](https://quantum-accelerators.github.io/quacc/dev/contributing.html#unit-tests).

### Notes

- Your PR will likely not be merged without proper and thorough tests.
- If you are an external contributor, you will see a comment from [@buildbot-princeton](https://github.com/buildbot-princeton). This is solely for the maintainers.
- When your code is ready for review, ping one of the [active maintainers](https://quantum-accelerators.github.io/quacc/about/contributors.html#active-maintainers).

## Summary of Changes

Most of this PR is code **deletion** to reflect the new changes on the ASE side, meaning no quacc.espresso.io and keys anymore.

To be able to test simple QE binaries I added a new job "post-processing" in core.py and I was able to add it and run it into the static_job_v2 test.

I suspect you will not like the new partial mess in write_input and read_output functions of the calculators, this was done with flexibility in mind, as most write/read functions in the future will likely be different and require different parameters. Same for _safe_io, we might want to add security measure in the future?

Some minor suggestions as well, like "prev_dir" to "prev_outdir" to be more consistent with QE keywords.

All tests pass on my side.

### CODE REVIEW

**Comment by buildbot-princeton** [comment]
Can one of the admins verify this patch?

**Review by Andrew-S-Rosen — COMMENTED** [review]
Thank you for your refactoring effort, @tomdemeyere. It is greatly appreciated.

The requested changes are minor. However, I don't quite understand the purpose of the `partial` functions in this context. I know from your message you said it's to proactively address future changes, but I don't quite see how they accomplish that goal. For now, I'd say just call the functions regularly, and if/when something breaks due to upstream changes, we can address that separately.

File: src/quacc/calculators/espresso/espresso.py [hunk 1]
```diff
@@ -80,17 +81,29 @@ def write_input(
...
+                **parameters,
+            )
+        elif self.binary == "ph":
+            write_espresso_ph(fd=fd, properties=properties, **parameters)
+        else:
+            write_fortran_namelist(
+                fd, binary=self.binary, properties=properties, **parameters
+            )
+
+        fd.close()
```
Comment (Andrew-S-Rosen) [inline src/quacc/calculators/espresso/espresso.py:106]: Can probably do `with Path.open(...)` so you don't have to explicitly `fd.close()`.

File: src/quacc/calculators/espresso/espresso.py [hunk 2]
```diff
@@ -110,9 +123,24 @@ def read_results(self, directory: Path | str) -> dict[str, Any]:
...
+                filename=directory / self.outputname,
+                format="espresso-out",
+                full_output=True,
+            )
+        elif self.binary == "ph":
+            results = read_espresso_ph(fd)
+        else:
+            results = {}
+
+        fd.close()
```
Comment (Andrew-S-Rosen) [inline src/quacc/calculators/espresso/espresso.py:139]: Same comment here about the context handler.

**Comment by Andrew-S-Rosen** [comment]
Thanks for your edits. Happy to merge once I get your okay. The only change (which I can easily do) is to make `prev_outdir` (`phonon_job`) consistent with `prev_dir` (`post_processing_job`).

**Comment by tomdemeyere** [comment]
I guess I overthought the initial thing, I thought using partial would allow more flexibility and save some line of codes. The _safe_io stuff was in case you wanted to add some additional security measure in the future (to avoid read and write on sensible data) as Ask mentioned a couple of times.

**Comment by Andrew-S-Rosen** [comment]
Thanks! To me, this all looks good and I have no concerns.

**Pull request merged**